target/linux/ramips/dts/mt7621_sztozed_zlt-s12-pro.dts

```diff
@@ -0,0 +1,256 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "sztozed,zlt-s12-pro", "mediatek,mt7621-soc";
+	model = "ZLT S12 PRO";
+
+	aliases {
+		led-boot = &led_power_yellow;
+		led-failsafe = &led_power_yellow;
+		led-running = &led_power_blue;
+		led-upgrade = &led_power_blue;
+		label-mac-device = &gmac0;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		modem-blue {
+			label = "blue:modem";
+			gpios = <&gpio 33 GPIO_ACTIVE_LOW>;
+		};
+
+		modem-red {
+			label = "red:modem";
+			gpios = <&gpio 27 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power_blue: power-blue {
+			label = "blue:power";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power_yellow: power-yellow {
+			label = "yellow:power";
+			gpios = <&gpio 29 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi {
+			label = "blue:wifi";
+			gpios = <&gpio 28 GPIO_ACTIVE_LOW>;
+		};
+
+		signal1 {
+			label = "blue:signal1";
+			gpios = <&gpio 25 GPIO_ACTIVE_LOW>;
+		};
+
+		signal2 {
+			label = "blue:signal2";
+			gpios = <&gpio 24 GPIO_ACTIVE_LOW>;
+		};
+
+		signal3 {
+			label = "blue:signal3";
+			gpios = <&gpio 23 GPIO_ACTIVE_LOW>;
+		};
+
+		signal4 {
+			label = "blue:signal4";
+			gpios = <&gpio 22 GPIO_ACTIVE_LOW>;
+		};
+
+		signal5 {
+			label = "blue:signal5";
+			gpios = <&gpio 26 GPIO_ACTIVE_LOW>;
+		};
+
+		phone {
+			label = "blue:phone";
+			gpios = <&gpio 31 GPIO_ACTIVE_LOW>;
+		};
+
+		slic_clk_switch {
+			label = "slic_clk_switch";
+			gpios = <&gpio 32 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	gpio_export {
+		compatible = "gpio-export";
+
+		lt72_ant0 {
+			gpio-export,name = "lt72_ant0";
+			gpio-export,input = <0>;
+			gpios = <&gpio 6 GPIO_ACTIVE_HIGH>;
+		};
+
+		lt72_ant1 {
+			gpio-export,name = "lt72_ant1";
+			gpio-export,input = <0>;
+			gpios = <&gpio 7 GPIO_ACTIVE_HIGH>;
+		};
+
+		lt72_power {
+			gpio-export,name = "lt72_power";
+			gpio-export,output = <1>;
+			gpios = <&gpio 14 GPIO_ACTIVE_HIGH>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	spidev@1 {
+		spi-cpol = <0x01>;
+		compatible = "rohm,dh2228fv";
+		spi-cpha = <0x01>;
+		linux,modalias = "spidev";
+		status = "okay";
+		reg = <0x01>;
+		spi-max-frequency = <0x1f4000>;
+	};
+
+	m25p80@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
```

Comment:
Could add support to uboot-envtools for reading this.

I *think* replacing label with compatible="u-boot,env" also works. I'm not entirely sure.

Reply:
> Could add support to uboot-envtools for reading this.
> 
> I _think_ replacing label with compatible="u-boot,env" also works. I'm not entirely sure.

This partition is not used by u-boot to store its env vars, it is empty.

Reply:
Right. That config needs to be done in a separate commit.

Last I tested it it didn't work. Would be great to get it working and get rid of that manual list.

Reply:
I'm not interested in working on this. `NVMEM_U_BOOT_ENV` was introduced by Rafal, I'm sure they're already cooking something regarding the uboot-envtools package. It would make even more sense to implement this on the main U-Boot repository where the uboot-envtools package retrieves the source code from. Till then, I'm going to focus on other things.

Reply:
Right. Different PR.